core/field_textinput.js

```diff
@@ -638,21 +619,15 @@ Blockly.FieldTextInput.prototype.getValueFromEditorText_ = function(text) {
  */
 Blockly.FieldTextInput.prototype.getScaledBBox = function() {
   if (this.fullBlockClickTarget_) {
-    var heightWidth = this.sourceBlock_.getHeightWidth();
+    var xy = this.getClickTarget_().getBoundingClientRect();
   } else {
-    var heightWidth = this.borderRect_.getBBox();
+    var xy = this.borderRect_.getBoundingClientRect();
```

Comment:
The trouble with the previous method (bbox + absoluteXY), is that absolute XY gets the size info of the click target / field svg root. This includes both the border rect and the text element. The text element (currently in geras) is slightly larger than the border rect which introduces the discrepancy when calculating positioning. The goal of sizing the text input is to match the dimensions of the border rect, hence why I'm switching it here to only use those dimensions.

Reply:
Cool.  Removing the workarounds certainly seems positive to me.